velox/core/QueryCtx.h

```diff
@@ -26,16 +26,20 @@
 
 namespace facebook::velox::core {
 
-class QueryCtx {
+class QueryCtx : public std::enable_shared_from_this<QueryCtx> {
```

Comment:
Can we separate the QueryCtx refactoring change from the main driver change? This PR looks dominated by this query ctx refactoring.